xilem_web/src/app.rs

```diff
@@ -1,44 +1,48 @@
 // Copyright 2023 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::{DomNode, ViewCtx};
 use std::{cell::RefCell, rc::Rc};
 
-use crate::{
-    context::Cx,
-    view::{DomNode, View},
-    Message,
-};
-use xilem_core::{Id, MessageResult};
+use crate::{DomView, DynMessage, PodMut};
+use xilem_core::{MessageResult, ViewId};
+
+type IdPath = Vec<ViewId>;
+
+pub(crate) struct AppMessage {
+    pub id_path: IdPath,
+    pub body: DynMessage,
+}
 
 /// The type responsible for running your app.
-pub struct App<T, V: View<T>, F: FnMut(&mut T) -> V>(Rc<RefCell<AppInner<T, V, F>>>);
+pub struct App<T, V: DomView<T>, F: FnMut(&mut T) -> V>(Rc<RefCell<AppInner<T, V, F>>>);
```

Comment:
I'd obviously much prefer this avoided interior mutability, but I'll assume it's necessary - I don't have enough context to know whether it's avoidable

Reply:
That's actually one of the few things from the old code-base that I haven't paid too much attention to, nor written it myself. But I do think it's necessary, since the `App` itself is "forgotten" to avoid blocking the event-loop, and just lives on within the view context (and message thunks).